sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -500,13 +506,52 @@ def _add_argparse_args(cls, parser):
                         choices=['COST_OPTIMIZED', 'SPEED_OPTIMIZED'],
                         help='Set the Flexible Resource Scheduling mode')
 
+  def __getattr__(self, name):
```

Comment:
Suggestion: consider using `@property` here.

Reply:
That was my first thought as well, but I couldn't figure out how to get it to work.

The problem is that attribute access of the `GoogleCloudOptions` class ends up being performed using `PipelineOptions.__getattr__` and `PipelineOptions.__setattr__`, and those only check if the attribute is set in the (child) classes `_add_argparse_args(cls, parser)` method, not if the (child) class actually has that attribute.

Since we can't have both `StandardOptions._add_argparse_args` and `GoogleCloudOptions._add_argparse_args` define the `temp_location` setting, I ended up patching the `GoogleCloudOptions.__getattr__` and `GoogleCloudOptions.__setattr__` methods to add an additional check for the `temp_location` attribute (and then call the parent `PipelineOptions.__getattr__` / `PipelineOptions.__setattr__`).